pyodide_build/common.py

```diff
@@ -43,17 +43,64 @@ def xbuildenv_dirname() -> str:
 @cache
 def default_xbuildenv_path() -> Path:
     """
-    Return the default path to the cross-build environment directory.
+    Return the default path to the cross-build environment directory. This directory
+    is used when no path is provided to `pyodide xbuildenv`'s subcommands.
 
-    This directory is used when no path is provided to the `pyodide xbuildenv` subcommands.
+    The search order for determining the path is as follows:
+    1. the PYODIDE_XBUILDENV_PATH environment variable if set
+    2. the value provided by "pyodide config xbuildenv_path"
+    3. the default location based on platformdirs
+
+    Returns
+    -------
+    Path
+        The path to the cross-build environment directory
     """
+
+    # 1. check environment variable
+    env_path = os.environ.get("PYODIDE_XBUILDENV_PATH")
+    if env_path:
+        path = Path(env_path)
+        if _has_write_access(path):
+            return path
+        else:
+            logger.warning(
+                "The directory specified in PYODIDE_XBUILDENV_PATH (%s) is not writable. "
+                "Falling back to other locations.",
+                path,
+            )
```

Comment:
Maybe we should fail here?

Reply:
I was thinking of it as well and was unsure, thanks for raising it – will change the warnings to errors.

Reply:
Addressed by c83f0ae3549f206fcefe02b161409d11f28f6de0.